extensions/2.0/Vendor/EXT_mesh_primitive_edge_visibility/README.md

```diff
@@ -0,0 +1,292 @@
+
+The `lineStrings` property **MUST** be defined if `visibility` is not defined - otherwise, the extension would encode no edges.
+
+#### Examples
+
+Consider the following simple triangle fan with triangles described by the index list `[0,2,1, 0,3,2, 0,4,3, 0,5,4]`:
+```
+  2______3______4
+  |＼    |    ／|
+  |  ＼  |  ／  |
+  |____＼|／____|
+  1      0      5
+```
+
+Assume that the top and bottom edges are hard edges, to be encoded as a pair of line strings with indices `[2,3,4]` and `[1,0,5]`, respectively; and the vertical edges are hard edges, to be encoded as simple line segments. Then the `visibility` bitfield should encode only the vertical edges, with visibility values `[0,2,0, 2,0,0, 0,0,3, 0,2,0]`.
+
+##### Single material
+
+Assume that the top and bottom edges are to be drawn using the same material, different from the material used to draw the vertical edges, and the indices are encoded as unsigned bytes. Then the index buffer would look like `[2,3,4, 255, 1,0,5]`, where `255` is the primitive restart value separating the two line strings. The `lineStrings` property would look something like this:
+```
+{
+    "lineStrings": [
+        {
+            "indices": 5,
+            "material": 1
+        }
+    ]
+}
+```
+
+##### Two materials
+
+Assume that the top edge is to be drawn using the same material as the vertical edges, and the bottom edge is to be drawn using a different material. Then there would be two index buffers - `[2,3,4]` and `[1,0,5]` - and the `lineStrings` property would look something like this:
+```
+{
+    "lineStrings": [
+        {
+            "indices": 5
+        },
+        {
+            "indices": 6,
+            "material": 1
+        }
+    ]
+}
+```
+
+### Silhouette Normals
+
+The extension's `silhouetteNormals` property specifies the index of an accessor of `SCALAR` type and component type 5123 (unsigned short) providing normal vectors used to determine the visibility of silhouette edges at display time. For each edge encoded as a silhouette (visibility value `1`) in `visibility`, the silhouette normals buffer provides the two outward-facing normal vectors of the pair of triangles sharing the edge. Each normal vector is compressed into 16 bits using an "oct16" encoding. The ordering of the normal vector pairs corresponds to the ordering of the edges in `visibility`; that is, the first pair of normals corresponds to the first edge encoded with visibility `1`, the second pair to the second occurrence of visibility `1`; and so on. The accessor's `count` **MUST** be twice the number of edges encoded with visibility value `1`.
+
+### oct16 encoding
+
+Octahedral normal encoding projects a normal vector onto an octahedron and folds it in half, mapping to two signed normalized components in [-1, 1]. The "oct16" form of this encoding maps each component to the integer range [0, 255], permitting a normal vector to be encoded into two bytes.
```

Comment:
Is there any reason not to use VEC2 accessor type with normalized BYTE component type?

It would enable implicit conversions from quantized values using the standard glTF rules. Also that component type is generally considered more accurate as preserves zeros.

Reply:
That would make sense. However I updated the spec with regards to the normal encoding based on feedback from @lilleyse and @pjcozzi.